components/channel_select/channel_select.tsx

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useState} from 'react';
+
+import Constants from 'utils/constants';
+import * as Utils from 'utils/utils.jsx';
+
+type Props = {
+    channels: Array<{
+        id: string;
+        display_name?: string;
+        name: string;
+        type: string;
+    }>;
+    onChange: () => void;
+    value: string;
```

Comment:
@devinbinnie 
`channels, onChange, and value` all seem to be optional in the original file. Should they be optional here or is having them required okay?

Reply:
They should be optional here as well unless they're always provided by redux. Good catch!